### PR TITLE
Fixes for Gaudi (> v30) compatibility [requires Gaudiv32r0]

### LINKED ISSUE #270: [v0.9] I/O doesn't work with Gaudiv30r1
Using LCG_92 and Gaudiv30r1, the root files written by PodioOutput do not contain all data.

### PR DESCRIPTION
**Pull request by vvolkl**
* Consistent use of macro `DECLARE_COMPONENTS` for Gaudi componentsm required by Gaudiv32
* DataService interface changed, thus the PodioDataService did not properly override the method `registerObject` anymore. Updated PodioDataSvc.
* Some Calorimeter tests stopped working with the new externals, but could be fixed by substituting `DectMaster` for the list of individual subdetectors. Still unclear what  caused this.
* The job that produces tracker hits for the `RecTracker` test stopped working with the new externals, but could be fixed by removing the `argparse` arguments and rewriting the job options file. Still unclear what exactly caused this

tested on ubuntu with Gaudiv32r0, won't work with the old Gaudi version, thus needs to be merged with new externals 

Fixes #270 
Supersedes #325

### CODE REVIEW

**Comment by vvolkl** [comment]
@phsft-bot build just on slc6/gcc8

**Comment by vvolkl** [comment]
@phsft-bot build just on slc6/gcc62

**Comment by JavierCVilla** [comment]
@phsft-bot build just on slc6/gcc62

**Comment by JavierCVilla** [comment]
@phsft-bot build just on slc6/gcc62

**Comment by JavierCVilla** [comment]
@phsft-bot build just on slc6/gcc62

**Comment by JavierCVilla** [comment]
@phsft-bot build just on slc6/gcc62

File: Generation/src/components/EDMToHepMCConverter.h [hunk 1]
```diff
@@ -12,7 +12,7 @@ class GenVertexCollection;
 }
 
 class EDMToHepMCConverter : public GaudiAlgorithm {
-  friend class AlgFactory<EDMToHepMCConverter>;
+  //friend class AlgFactory<EDMToHepMCConverter>;
```
Comment (JavierCVilla) [inline Generation/src/components/EDMToHepMCConverter.h]: Remove comment.

File: Examples/src/CreateSampleJet.h [hunk 2]
```diff
@@ -10,7 +10,7 @@ class JetCollection;
 }
 
 class CreateSampleJet : public GaudiAlgorithm {
-  friend class AlgFactory<CreateSampleJet>;
+  //friend class AlgFactory<CreateSampleJet>;
```
Comment (JavierCVilla) [inline Examples/src/CreateSampleJet.h]: Remove comment

File: FWCore/tests/scripts/check_coll_after_read.py [hunk 3]
```diff
@@ -2,19 +2,15 @@
 from EventStore import EventStore
 
 gSystem.Load("libdatamodelDict")
-store = EventStore(["./pythia_test.root"])
+#store = EventStore(["./pythia_test.root"])
```
Comment (JavierCVilla) [inline FWCore/tests/scripts/check_coll_after_read.py]: Remove comment

File: Generation/src/components/GenAlg.h [hunk 4]
```diff
@@ -17,7 +17,7 @@ class GenEvent;
 }
 
 class GenAlg : public GaudiAlgorithm {
-  friend class AlgFactory<GenAlg>;
+  //friend class AlgFactory<GenAlg>;
```
Comment (JavierCVilla) [inline Generation/src/components/GenAlg.h]: Remove comment

File: Generation/src/components/GenMerge.h [hunk 5]
```diff
@@ -25,7 +25,7 @@ class MCParticleCollection;
  *  Associations between particles and vertices are updated, so it is safe to drop the old collections.
  */
 class GenMerge : public GaudiAlgorithm {
-  friend class AlgFactory<GenMerge>;
+  //friend class AlgFactory<GenMerge>;
```
Comment (JavierCVilla) [inline Generation/src/components/GenMerge.h]: Remove comment

File: Generation/src/components/GenParticleFilter.h [hunk 6]
```diff
@@ -19,7 +19,7 @@ class MCParticleCollection;
 */
 
 class GenParticleFilter : public GaudiAlgorithm {
-  friend class AlgFactory<GenParticleFilter>;
+  //friend class AlgFactory<GenParticleFilter>;
```
Comment (JavierCVilla) [inline Generation/src/components/GenParticleFilter.h]: Remove comment

File: Generation/src/components/HepEVTReader.h [hunk 7]
```diff
@@ -37,7 +37,7 @@ class GenVertexCollection;
  */
 
 class HepEVTReader: public GaudiAlgorithm {
-  friend class AlgFactory<HepEVTReader> ;
+  //friend class AlgFactory<HepEVTReader> ;
```
Comment (JavierCVilla) [inline Generation/src/components/HepEVTReader.h]: Remove comment

File: Generation/src/components/HepMCDumper.h [hunk 8]
```diff
@@ -8,7 +8,7 @@
 #include "HepMC/GenEvent.h"
 
 class HepMCDumper : public GaudiAlgorithm {
-  friend class AlgFactory<HepMCDumper>;
+  //friend class AlgFactory<HepMCDumper>;
```
Comment (JavierCVilla) [inline Generation/src/components/HepMCDumper.h]: Remove comment

File: Generation/src/components/HepMCFileWriter.h [hunk 9]
```diff
@@ -21,7 +21,7 @@ class IO_GenEvent;
  */
 
 class HepMCFileWriter : public GaudiAlgorithm {
-  friend class AlgFactory<HepMCFileWriter>;
+  //friend class AlgFactory<HepMCFileWriter>;
```
Comment (JavierCVilla) [inline Generation/src/components/HepMCFileWriter.h]: Remove comment

File: Generation/src/components/HepMCHistograms.h [hunk 10]
```diff
@@ -9,7 +9,7 @@
 #include "TH1F.h"
 
 class HepMCHistograms : public GaudiAlgorithm {
-  friend class AlgFactory<HepMCHistograms>;
+  //friend class AlgFactory<HepMCHistograms>;
```
Comment (JavierCVilla) [inline Generation/src/components/HepMCHistograms.h]: Remove comment

File: Generation/src/components/HepMCToEDMConverter.h [hunk 11]
```diff
@@ -12,7 +12,7 @@ class GenVertexCollection;
 }
 
 class HepMCToEDMConverter : public GaudiAlgorithm {
-  friend class AlgFactory<HepMCToEDMConverter>;
+  //friend class AlgFactory<HepMCToEDMConverter>;
```
Comment (JavierCVilla) [inline Generation/src/components/HepMCToEDMConverter.h]: Remove comment

File: Reconstruction/RecCalorimeter/src/components/HepMCJetClustering.h [hunk 12]
```diff
@@ -9,7 +9,7 @@
 #include "fastjet/JetDefinition.hh"
 
 class HepMCJetClustering : public GaudiAlgorithm {
-  friend class AlgFactory<HepMCJetClustering>;
+  //friend class AlgFactory<HepMCJetClustering>;
```
Comment (JavierCVilla) [inline Reconstruction/RecCalorimeter/src/components/HepMCJetClustering.h]: Remove comment

File: Reconstruction/RecCalorimeter/src/components/JetClustering.h [hunk 13]
```diff
@@ -20,7 +20,7 @@
 template <class P, class J>
 class JetClustering : public GaudiAlgorithm {
   // Colin: the following doesn't seem to be necessary
-  // friend class AlgFactory< JetClustering<P, J> > ;
+  // //friend class AlgFactory< JetClustering<P, J> > ;
```
Comment (JavierCVilla) [inline Reconstruction/RecCalorimeter/src/components/JetClustering.h]: Remove comment

File: Reconstruction/RecCalorimeter/src/components/JetHistograms.h [hunk 14]
```diff
@@ -9,7 +9,7 @@
 #include "TH1F.h"
 
 class JetHistograms : public GaudiAlgorithm {
-  friend class AlgFactory<JetHistograms>;
+  //friend class AlgFactory<JetHistograms>;
```
Comment (JavierCVilla) [inline Reconstruction/RecCalorimeter/src/components/JetHistograms.h]: Remove comment

File: Reconstruction/RecCalorimeter/src/components/NoiseCaloCellsFromFileTool.h [hunk 15]
```diff
@@ -4,7 +4,9 @@
 // from Gaudi
 #include "GaudiAlg/GaudiTool.h"
 #include "GaudiKernel/RndmGenerators.h"
-class IRndmGenSvc;
+#include "GaudiKernel/IRndmGenSvc.h"
+
+//class IRndmGenSvc;
```
Comment (JavierCVilla) [inline Reconstruction/RecCalorimeter/src/components/NoiseCaloCellsFromFileTool.h]: Remove comment

File: Reconstruction/RecCalorimeter/tests/options/runBarrelCaloSystem_ReconstructionTopoClusters_noNoise.py [hunk 16]
```diff
@@ -28,9 +28,9 @@
                         OutputLevel = DEBUG)
 
 from Configurables import GeoSvc 
-detectors_to_use =['file:Detector/DetFCChhBaseline1/compact/FCChh_DectEmptyMaster.xml',
-                   'file:Detector/DetFCChhECalInclined/compact/FCChh_ECalBarrel_withCryostat.xml',
-                   'file:Detector/DetFCChhHCalTile/compact/FCChh_HCalBarrel_TileCal.xml'
+detectors_to_use =['file:Detector/DetFCChhBaseline1/compact/FCChh_DectMaster.xml',
+                   #'file:Detector/DetFCChhECalInclined/compact/FCChh_ECalBarrel_withCryostat.xml',
```
Comment (JavierCVilla) [inline Reconstruction/RecCalorimeter/tests/options/runBarrelCaloSystem_ReconstructionTopoClusters_noNoise.py]: Are these commented files still required?
Does it work with them?

File: Reconstruction/RecTracker/options/TrickTrackReco.py [hunk 17]
```diff
@@ -133,17 +133,17 @@
...
+riemann_fit = RiemannFitTool("TheRiemannFitTool")
 
 from Configurables import RecTrackAlg
 RecTrackAlg = RecTrackAlg()
-if args.truthseeding:
-  RecTrackAlg.TrackSeedingTool = truth_seeds
-else:
-  RecTrackAlg.TrackSeedingTool = tricktrack_seed_tool
-RecTrackAlg.TrackerPositionedHits.Path = collectionNames["TrackerPositionedHits"]
+#RecTrackAlg.TrackSeedingTool = truth_seeds
```
Comment (JavierCVilla) [inline Reconstruction/RecTracker/options/TrickTrackReco.py]: Can this be removed?

File: Sim/SimDelphesInterface/src/DelphesSimulation.h [hunk 18]
```diff
@@ -53,7 +53,7 @@ class TObjArray;
  */
 class DelphesSimulation : public GaudiAlgorithm {
 
-  friend class AlgFactory<DelphesSimulation>;
+  //friend class AlgFactory<DelphesSimulation>;
```
Comment (JavierCVilla) [inline Sim/SimDelphesInterface/src/DelphesSimulation.h]: Remove comment

File: Test/TestGeneration/src/components/HepMCFromScratch.h [hunk 19]
```diff
@@ -9,7 +9,7 @@ class GenEvent;
 }
 
 class HepMCFromScratch : public GaudiAlgorithm {
-  friend class AlgFactory<HepMCFromScratch>;
+  //friend class AlgFactory<HepMCFromScratch>;
```
Comment (JavierCVilla) [inline Test/TestGeneration/src/components/HepMCFromScratch.h]: Remove comment

File: init.sh [hunk 20]
```diff
@@ -28,13 +28,14 @@ if test -f "/etc/redhat-release"; then
...
 # Compiler
-gcctag="gcc62"
+gcctag="gcc8"
 
 # Platform
 platform="$ostag-$gcctag-opt"
 
 # Setup script
-setuppath="/cvmfs/fcc.cern.ch/sw/views/releases/externals/$versiontag/$platform/setup.sh"
+#setuppath="/cvmfs/fcc.cern.ch/sw/views/releases/externals/$versiontag/$platform/setup.sh"
```
Comment (JavierCVilla) [inline init.sh]: This needs to be changed once we install a new release of the fcc-externals.

**Comment by vvolkl** [comment]
Thanks for the comments Javier. Yes, please let me know as soon as there is a release of the fcc externals v96 available on cvmfs

**Comment by JavierCVilla** [comment]
@phsft-bot build

**Comment by JavierCVilla** [comment]
@phsft-bot build just on slc6/gcc62

**Comment by vvolkl** [comment]
Not sure why the CI didn't build on centos7, but the PR should be ok.

**Comment by gganis** [comment]
@phsft-bot build just on ROOT-centos7/gcc8

**Comment by gganis** [comment]
@phsft-bot build just on centos7/gcc8

**Comment by vvolkl** [comment]
@phsft-bot build just on centos7/gcc8

**Comment by vvolkl** [comment]
@phsft-bot build just on centos7/gcc8

**Pull request merged**